source/features/locked-issue-indicator.tsx

```diff
@@ -0,0 +1,60 @@
+import './locked-issue-indicator.css';
+
+import React from 'react';
+import {LockIcon} from '@primer/octicons-react';
+import * as pageDetect from 'github-url-detection';
+
+import {$} from 'select-dom';
+
+import features from '../feature-manager.js';
+import observe from '../helpers/selector-observer.js';
+import {isHasSelectorSupported} from '../helpers/select-has.js';
+
+function LockedIndicator(): JSX.Element {
+	return (
+		<span title="Locked" className="State d-flex flex-items-center">
+			<LockIcon className="flex-items-center mr-1"/>
+			Locked
+		</span>
+	);
+}
+
+function addLock(): void {
+	// Add locked indicator to header
+	$('.gh-header-meta > :first-child')!.after(
+		<div className="flex-shrink-0 mb-2 flex-self-start flex-md-self-center rgh-locked-issue-indicator">
+			<LockedIndicator/>
+		</div>,
+	);
+	// Add locked indicator to sticky header
+	$('.gh-header-sticky .flex-row > :first-child')!.after(
+		<div className="mr-2 mb-2 flex-shrink-0 rgh-locked-issue-indicator">
+			<LockedIndicator/>
+		</div>,
+	);
+}
+
+function init(signal: AbortSignal): void {
+	// If reactions-menu exists, then .js-pick-reaction is the second child
+	observe(':has(.js-pick-reaction:first-child) .gh-header-meta', addLock, {signal});
```

Comment:
`observe` acts as a selector, we should not need to also use `$` in `addLock`
```suggestion
	observe(':has(.js-pick-reaction:first-child) .gh-header-meta > :first-child', addLock, {signal});
	observe(':has(.js-pick-reaction:first-child) .gh-header-sticky .flex-row > :first-child', addLockToSticky, {signal});
```